### PR TITLE
feat: port media/codecs to CMake

### PR DESCRIPTION
**Pull request by cadubentzen**
- [x] Compile media/codecs with CMake
- [x] Compile tests for media/codecs with CMake
- [x] Pass tests for media/codecs with CMake

Also:
- [x] Use the same clang-format  version as the CI here

Related to #1047

### CODE REVIEW

File: packager/media/base/raw_key_source.cc [hunk 1]
```diff
@@ -56,9 +57,9 @@ Status RawKeySource::GetKey(const std::vector<uint8_t>& key_id,
       return Status::OK;
     }
   }
-  std::string key_id_str(key_id.begin(), key_id.end());
   return Status(error::INTERNAL_ERROR,
-                "Key for key_id=" + absl::BytesToHexString(key_id_str) +
+                "Key for key_id=" +
+                    absl::BytesToHexString(byte_vector_to_string_view(key_id)) +
```
Comment (cadubentzen) [inline packager/media/base/raw_key_source.cc:62]: One pattern that repeated many times was changing `base::HexEncode(data*,size) ->  absl::BytesToHexString(string_view)`

The refactorings so far were done like here, but the problem is that it performs an extra copy+allocation.
What worked for me was constructing a string_view directly from the uint8_t array/vector. Sadly, with string_view the constructors can't take in a uint8_t* so a `reinterpret_cast` is needed. Created helper functions for it.

File: packager/media/codecs/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 Bitmovin Inc. All rights reserved.
```
Comment (joeyparrish) [inline packager/media/codecs/CMakeLists.txt:1]: Please put "Google LLC" in the copyright header.  This is consistent with Google's requirement that contributors sign a CLA.

(I don't make these rules.)

File: packager/utils/bytes_to_string_view.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 Bitmovin Inc. All rights reserved.
```
Comment (joeyparrish) [inline packager/utils/bytes_to_string_view.h:1]: Same here: Google LLC, please.

File: packager/utils/bytes_to_string_view.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+#ifndef PACKAGER_UTILS_BYTES_TO_STRING_VIEW_H_
+#define PACKAGER_UTILS_BYTES_TO_STRING_VIEW_H_
+
+#include <string_view>
+#include <vector>
+
+namespace shaka {
+
+/// Convert byte array to string_view
```
Comment (joeyparrish) [inline packager/utils/bytes_to_string_view.h:15]: Well done!  I appreciate this.

**Review by joeyparrish — CHANGES_REQUESTED** [review]

File: packager/media/codecs/CMakeLists.txt (hunk 2 above)
Comment (cadubentzen) [reply packager/media/codecs/CMakeLists.txt:1]: No problem. Can I add "Google LLC" _in addition_ to Bitmovin or must I replace it?

**Comment by zdanek** [comment]
@cadubentzen can we go with this PR? What is blocking us? 
I depend on it in two PRs :)

**Comment by cadubentzen** [comment]
> @cadubentzen can we go with this PR? What is blocking us? I depend on it in two PRs :)

Hi @zdanek sorry for the delay here. At this point, only the corporate CLA is blocking it. I have already pinged folks internally to double-check it but unfortunately, it's not solved yet and I can't really speed it up :(

**Comment by zdanek** [comment]
> only the corporate CLA is blocking it
that's why I'm here as an individual playing at my own rules in spare time :)

But, yeah, I do understand. I still have a lot to do in `mpd`. 
Cheers.

**Comment by joeyparrish** [comment]
@cadubentzen, the CLA bot is happy with you, so that's good enough for me.  Can you fix the copyright headers (or let me know that you don't mind me doing it)?  I'm ready to merge this once that is resolved.

**Comment by cadubentzen** [comment]
> @cadubentzen, the CLA bot is happy with you, so that's good enough for me. Can you fix the copyright headers (or let me know that you don't mind me doing it)? I'm ready to merge this once that is resolved.

@joeyparrish Thanks, just fixed the copyright headers. Could you please re-trigger CI here?

**Review by joeyparrish — APPROVED** [review]

**Comment by joeyparrish** [comment]
Looks like there are some build errors.  At least one was a missing header.  Can you please take a look?

**Comment by joeyparrish** [comment]
```
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(192,17): error C2220: the following warning is treated as an error [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(192,17): warning C4459: declaration of 'kSps' hides global declaration [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(17,15): message : see declaration of 'shaka::media::`anonymous-namespace'::kSps' [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(212,17): warning C4459: declaration of 'kSps' hides global declaration [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(17,15): message : see declaration of 'shaka::media::`anonymous-namespace'::kSps' [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(233,17): warning C4459: declaration of 'kSps' hides global declaration [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(17,15): message : see declaration of 'shaka::media::`anonymous-namespace'::kSps' [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(259,17): warning C4459: declaration of 'kSps' hides global declaration [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\packager\media\codecs\h264_parser_unittest.cc(17,15): message : see declaration of 'shaka::media::`anonymous-namespace'::kSps' [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
  h265_byte_to_unit_stream_converter_unittest.cc
  h265_parser_unittest.cc
  h26x_bit_reader_unittest.cc
  hevc_decoder_configuration_record_unittest.cc
  hls_audio_util_unittest.cc
  nal_unit_to_byte_stream_converter_unittest.cc
D:\a\shaka-packager\shaka-packager\.\packager/media/formats/mp4/box_definitions_comparison.h(350,51): error C2220: the following warning is treated as an error [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\.\packager/media/formats/mp4/box_definitions_comparison.h(350,51): warning C4100: 'rhs': unreferenced formal parameter [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\.\packager/media/formats/mp4/box_definitions_comparison.h(349,51): warning C4100: 'lhs': unreferenced formal parameter [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\.\packager/media/formats/mp4/box_definitions_comparison.h(491,73): warning C4100: 'rhs': unreferenced formal parameter [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
D:\a\shaka-packager\shaka-packager\.\packager/media/formats/mp4/box_definitions_comparison.h(491,46): warning C4100: 'lhs': unreferenced formal parameter [D:\a\shaka-packager\shaka-packager\build\packager\media\codecs\codecs_unittest.vcxproj]
```

**Comment by cadubentzen** [comment]
@joeyparrish I'm also now checking in parallel with the CI builds at my fork. Will ping you once the build is green there so you can retry it here :slightly_smiling_face:

**Comment by joeyparrish** [comment]
See also https://github.com/shaka-project/shaka-packager/pull/1157, which fixes the Linux release builds.  (If you'll review it, I'll go ahead and merge it today.)

**Comment by cadubentzen** [comment]
@joeyparrish it should be good to go now :slightly_smiling_face: Got a green CI [here](https://github.com/cadubentzen/shaka-packager/actions/runs/3708282122).

**Comment by joeyparrish** [comment]
Running it again now.  BTW, I am working on ways to speed up CI builds so you can have a tighter feedback loop in situations like this.  It might take some time for me to get it right, but I'm working on it.

**Review by joeyparrish — APPROVED** [review]

**Pull request merged**